incubator/sparkoperator/templates/spark-operator-deployment.yaml

```diff
@@ -53,27 +53,24 @@ spec:
         {{- end }}
         {{- if .Values.enableMetrics }}
         ports:
-          - containerPort: 10254
+          - containerPort: {{ int .Values.metricsPort }}
         {{ end }}
         command: ["/usr/bin/spark-operator"]
-        {{- if and (.Values.enableMetrics) (.Values.enableWebhook) }}
-        args:
-        - -enable-metrics=true
-        - -metrics-labels=app_type
-        - -enable-webhook=true
-        - -webhook-svc-namespace={{ .Release.Namespace }}
-        - -logtostderr
-        {{- else if (.Values.enableMetrics) }}
         args:
+        {{- if .Values.enableMetrics }}
         - -enable-metrics=true
         - -metrics-labels=app_type
-        - -logtostderr
-        {{- else if (.Values.enableWebhook) }}
-        args:
+        - -metrics-port={{ .Values.metricsPort }}
+        - -metrics-endpoint={{ .Values.metricsEndpoint }}
+        - -metrics-prefix={{ .Values.metricsPrefix }}
+        {{- end }}
+        {{- if .Values.enableWebhook }}
         - -enable-webhook=true
         - -webhook-svc-namespace={{ .Release.Namespace }}
-        - -logtostderr
-        {{- else }}
-        args:
-        - -logtostderr
+        - -webhook-port={{ .Values.webhookPort }}
         {{- end }}
+        - -install-crds={{ .Values.installCrds }}
+        - -controller-threads={{ .Values.controllerThreads }}
+        - -submission-threads={{ .Values.submissionThreads }}
```

Comment:
Just realize that PR https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/291 will get rid of flag `submission-threads`, so we might want to remove it here.